merge unit tests to master

LauchInterceptor opened the pull request.

LauchInterceptor committed:
added unit test

jo2 commented:
There are some problems with the tests:
DependencyTreeApplicationTests line 65: Files which are no source files (".java") are not regarded. So the amount of child nodes can't be equal to the amount of files there.

DependencyTreeApplicationTests line 69: Empty directories are not regarded by git. there is no package "emptypackage" in the testSrc directory.

DependencyTreeApplicationTests line 123: CoreTest has a dependency on CoreDependency but the file is named CoreDependencyTest so the dependency can't be resoolved.

DependencyTreeApplicationTests line 127: Same reason as above. CoreDependency is not seen as a dependecy so CoreTest has only 3 dependencies.

DependencyTreeApplicationTests line 137: DependencyTree needs valid code for its analysis. import statements aren't valid. We have to discuss if the code should be checked for compilation errors before it is analysed or if these invalid dependencies should not be found as dependencies by the regex patterns. Right now the check for import statements ends if a line is not either of an import statement, an empty line, a comment or the package declaration.

DependencyTreeApplicationTests line 115: The compare function in checkOrder is not working correctly. If o1 has a dependency on o2, o1 should be in the list before o2.

LauchInterceptor committed:
Merge remote-tracking branch 'origin/master' into lorenz-tests

LauchInterceptor committed:
add dummy file in emptypackage, fixed InvalidDependencyTest and CoreTest

added dummy file to emptypackage so its included with by git, fixed InvalidDependencyTest to match specification in regards to analyzing error free code only, fixed a refractoring mistake in CoreTest

LauchInterceptor committed:
fixed test project path and test behaviour

test project path is now local to workspace

LauchInterceptor committed:
Merge remote-tracking branch 'origin/master' into lorenz-tests

LauchInterceptor committed:
add test for fully classified dependencies

LauchInterceptor commented:
Please review the test again
